### PR TITLE
Fixes typo in docs for vis.addOverlay and refactors it

### LINKED ISSUE #405: Improve the documentation for cartodb.geo.ui.Tooltip and refactor
The example in [cartodb.geo.ui.Tooltip](https://github.com/CartoDB/cartodb.js/blob/develop/doc/API.md#cartodbgeouitooltip) is missing a comma after the `type` parameter. Also, it would be nice to specify that someone can pass a `layer` or the first layer of the visualization will be used.

[The implementation could be improved too](https://github.com/CartoDB/cartodb.js/blob/develop/src/vis/overlays.js#L430-L440). We could raise the exception at the beginning of the method if no layer has been passed and the visualization has no layers.

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #405

This one is for you @xavijam.

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -826,7 +826,7 @@ Shows a small tooltip on hover:
 <div class="code-title">cartodb.geo.ui.Tooltip</div>
 ```javascript
 var tooltip = vis.addOverlay({
-  type: 'tooltip'
+  type: 'tooltip',
```
Comment (xavijam) [inline doc/API.md]: I'd add that layer is optional.

File: src/vis/overlays.js [hunk 2]
```diff
@@ -426,23 +426,12 @@ cdb.vis.Overlay.register('search', function(data, vis) {
...
+  if (!data.layer && vis.getLayers().length <= 1) {
     throw new Error("layer is null");
   }
-
+  data.layer = data.layer || vis.getLayers()[1];
   data.layer.setInteraction(true);
-  var tooltip = new cdb.geo.ui.Tooltip(data);
-  return tooltip;
-
+  return new cdb.geo.ui.Tooltip(data);;
```
Comment (xavijam) [inline src/vis/overlays.js]: double semicolon

Comment (alonsogarciapablo) [inline src/vis/overlays.js]: good catch!

File: doc/API.md [hunk 3]
```diff
@@ -231,7 +231,7 @@ Adds an overlay to the map that can be either a zoom control, a tooltip or an in
   - **layer** layer from the visualization where the overlay should be applied (optional)
   - **type** zoom / tooltip / infobox
 
-Extra options are available based on the UI component selected before.
+If no layer is provided, the overlay will be added to the first layer of the visualization. Extra options are available based on the specific UI component.
```
Comment (alonsogarciapablo) [inline doc/API.md]: @xavijam I've added this here to explain what happens if you use vis.addOverlay without a layer.

**Comment by alonsogarciapablo** [comment]
@xavijam If you're cool with this I will merge it.

**Comment by xavijam** [comment]
Looks good :+1:

**Pull request merged**